script.js

```diff
@@ -1,4 +1,348 @@
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+        cardValue = 1;
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardValue = 10;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        score: cardValue,
+      };
+
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+// SHUFFLE DECK
+//
+//
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+//START MAIN FUNCTION
+//
+//
+var shuffledDeck = shuffleCards(makeDeck());
+var playerDeck = shuffledDeck.splice(0, 26);
+var computerDeck = shuffledDeck;
+var currentGameMode = "draw card";
+
+var playerCards = [];
+var computerCards = [];
+console.log("player cards: ");
+console.log(playerCards);
+console.log("computer cards: ");
+console.log(computerCards);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var playerScore = 0;
+  var computerScore = 0;
+
+  // Create a function to calcualte player's total score.
+  var calculatePlayerScore = function () {
+    var playerIndex = 0;
+    while (playerIndex < playerCards.length) {
+      playerScore = playerScore + playerCards[playerIndex].score;
+      playerIndex = playerIndex + 1;
+    }
+    if (playerCards[0].score == 10 && playerCards[1].name == "ace") {
+      playerScore = 21;
+    }
+    if (playerCards[1].score == 10 && playerCards[0].name == "ace") {
+      playerScore = 21;
+    }
```

Comment:
Hmm. The calculation here is a little tricky. If I draw an Ace then a 9, should my current score be a 20 (assuming 11 + 9), or a 10 (assuming 1 + 9)?

The logic written here assumes that Aces are only counted as 11 only if Ace and a card with a score of 10 appears in the first 2 cards in `playerCards`. What happens in the situation where Ace appears in the 3rd card, or 4th card when you hit?

It seems like in all other situations, Aces remain as 1. It will still work in your demo later, but it is restrictive in flexibility: what if I draw a 3 and a 7 initially (first 2 cards), and then I draw an Ace as my third card (when I hit), and choose to stand after drawing the third card? My final score will be 11 with your current logic, when it should also be interpreted as 21 too.